### PR TITLE
docs: update tanstack query integration documentation to the new api

### PR DESCRIPTION
**Pull request by kalijonn**
## Related Issues or Discussions

Related: https://github.com/jotaijs/jotai-tanstack-query/pull/45

## Check List

- [x] `yarn run prettier` for formatting code and docs

### CODE REVIEW

**Comment by kalijonn** [comment]
I'll update csbs after releasing v.0.8.0

**Review by dai-shi — COMMENTED** [review]
I think we need migration guide to v0.8.0.
Either a section in this doc, or a link to a doc in jotai-tanstack-query repo is fine.

**Comment by kalijonn** [comment]
> I think we need migration guide to v0.8.0. Either a section in this doc, or a link to a doc in jotai-tanstack-query repo is fine.

I added a migration guide. Can you review the level of detail? Too detailed or not detailed enough?

I also released jotai-tanstack-query v0.8.0 and added updated the sandboxes here.

File: docs/extensions/query.mdx [hunk 1]
```diff
@@ -7,53 +7,69 @@ keywords: tanstack,query
...
+`atomWithQuery` creates a new atom that implements a standard [`Query`](https://tanstack.com/query/v5/docs/react/guides/queries) from TanStack Query.
 
 ```jsx
 import { atom, useAtom } from 'jotai'
-import { atomsWithQuery } from 'jotai-tanstack-query'
+import { atomWithQuery } from 'jotai-tanstack-query'
 
 const idAtom = atom(1)
-const [userAtom] = atomsWithQuery((get) => ({
+const userAtom = atomsWithQuery((get) => ({
```
Comment (dai-shi) [inline docs/extensions/query.mdx:72]:
```suggestion
const userAtom = atomWithQuery((get) => ({
```

File: docs/extensions/query.mdx [hunk 2]
```diff
@@ -62,91 +78,169 @@ const [userAtom] = atomsWithQuery((get) => ({
...
 
-### `atomsWithMutation` usage
+### atomWithMutation usage
 
-`atomsWithMutation` creates new atoms that implement a standard [`Mutation`](https://tanstack.com/query/v4/docs/guides/mutations) from TanStack Query.
+`atomWithMutation` creates a new atom that implements a standard [`Mutation`](https://tanstack.com/query/v5/docs/guides/mutations) from TanStack Query.
 
 > Unlike queries, mutations are typically used to create/update/delete data or perform server side-effects.
 
 ```jsx
```
Comment (dai-shi) [inline docs/extensions/query.mdx:140]:
```suggestion
```tsx
```

**Review by dai-shi — COMMENTED** [review]
I think the migration guide is good.

**Review by dai-shi — APPROVED** [review]

**Pull request merged**